File: services/api/src/restaurant_os_api/modules/restaurant/domain/entities/menu_item_availability.py
"""MenuItemAvailability entity.

Restaurant Platform Architecture SS3.1: the availability-dimension twin
of ``MenuItemBranchPrice`` -- a branch- and time-scoped override of
``menu_items.is_available``. ``effective_to = None`` is an open-ended
86 until manually cleared (the Blueprint's "86 List Management" screen).
"""

from __future__ import annotations

from dataclasses import dataclass
from datetime import datetime


@dataclass(slots=True)
class MenuItemAvailability:
    id: str
    tenant_id: str
    branch_id: str
    menu_item_id: str
    is_available: bool
    effective_from: datetime
    created_at: datetime
    effective_to: datetime | None = None

File: apps/admin-web/src/types/menu-item-availability.ts
/**
 * Mirrors modules/restaurant/presentation/schemas/
 * menu_item_availability_schemas.py -- the availability-dimension twin
 * of MenuItemBranchPrice; see that type's own docstring for the flat
 * route / history-not-resolution reasoning, which applies identically
 * here.
 */

export interface MenuItemAvailability {
  id: string
  tenantId: string
  branchId: string
  menuItemId: string
  isAvailable: boolean
  effectiveFrom: string
  effectiveTo: string | null
  createdAt: string
}

export interface CreateMenuItemAvailabilityRequest {
  branchId: string
  isAvailable: boolean
  effectiveFrom: string
  effectiveTo?: string | null
}
